src/main/java/appeng/parts/networking/PartQuartzFiber.java

```diff
@@ -180,26 +220,29 @@ public double injectAEPower( final double amt, final Actionable mode, final Set<
 			final IEnergyGrid eg = this.outerProxy.getEnergy();
 			if( !seen.contains( eg ) )
 			{
-				return eg.injectAEPower( amt, mode, seen );
+				amount = eg.injectProviderPower( amount, mode, seen );
 			}
 		}
 		catch( final GridAccessException e )
 		{
 			// :P
 		}
 
-		return amt;
+		return amount;
 	}
 
 	@Override
-	public double getEnergyDemand( final double amt, final Set<IEnergyGrid> seen )
+	public double getProviderEnergyDemand( final double amt, final Set<IEnergyGridProvider> seen )
 	{
 		double demand = 0;
 
 		try
 		{
 			final IEnergyGrid eg = this.getProxy().getEnergy();
-			demand += eg.getEnergyDemand( amt - demand, seen );
+			if( !seen.contains( eg ) )
+			{
+				demand += eg.getProviderEnergyDemand( amt - demand, seen );
+			}
 		}
 		catch( final GridAccessException e )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)